gdsfactory/typings.py

```diff
@@ -215,14 +215,15 @@ class Routes(BaseModel):
     model_config = {"extra": "forbid"}
 
 
-class ComponentModel(BaseModel):
-    component: str | dict[str, Any]
-    settings: dict[str, Any] | None
+class Instance(BaseModel):
+    component: ComponentSpec
```

Comment:
I would argue that we should probably have a `Component` model also that is basically the component name + settings, rather than let this be any of the many possibilities allowed by `ComponentSpec`. we can still make a function to convert a `ComponentSpec` to a Component model, and allow a ComponentSpec to be passed to the `add_instance()` method

Reply:
or alternatively we replace the `component` field here with `component_name` and `component_settings` fields. but if we might want to add things in the future, such as component info, and the name of the parameterized component vs the function, etc., I think it's probably better to have a Component model, agreed?